CodeComplexity/src/TimeComplexity.cpp

```diff
@@ -0,0 +1,244 @@
+/**
+Here you will be given 14 functions for which you have to return Big-Oh time complexity as a string.
+You only need to fill return values for functions whose name starts with "function", it might use some helper functions.
+You need to calculate the complexity of those helper functions too and return overall bigoh complexity from
+wrapper function.
+
+For example for function0. You need to return "O(n)";
+
+If exacty time complexity is 6(n*n) . You just need to return n*n.
+Rules for returning Strings :
+'O' should be capital.
+You can't write nlogn, write O(n*logn).
+Use ^ character only if the power is not a constant. 
+n^2 is to be written as n*n.
+n^n can be written as it is.
+
+For functions taking arrays as input ,consider n as size of array.
+
+Try to solve the tasks without looking at answers, even if testcases fail.This makes sure you are learning really.
+*/
+
+#include<stdio.h>
+#include<math.h>
+
+char* function0(int n) {
+	//Sample Not tested
+	int i, j;
+	for (i = 0; i < n; i++) {
+
+	}
+
+	return "O(n)";
```

Comment:
I asked to write reasoning for each of the code complexity. Please write the comments why do you think that whatever the value you are returning in each function is the code complexity for the given code.